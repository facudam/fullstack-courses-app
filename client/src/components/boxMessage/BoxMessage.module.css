.box-message {
    padding: 7px 10px;
    font-size: 14px;
    background-color: var(--blue-hover);
    border-radius: 6px;
    color: #fff;
    text-shadow: 1px 1px 1px #000;
}