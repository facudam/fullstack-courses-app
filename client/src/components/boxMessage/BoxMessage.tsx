import { FC } from "react";
import styles from './BoxMessage.module.css'

interface Props {
    message: string
}
export const BoxMessage: FC<Props> = ({ message }) => {
    return (
        // El data-testid es para los tests.
        <div data-testid="test-message" className={ styles['box-message']}>
            <p>{ message }</p>
        </div> 
    )
}
