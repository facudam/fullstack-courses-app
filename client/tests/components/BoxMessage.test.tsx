import { render, screen } from '@testing-library/react';
import { BoxMessage } from '../../src/components/boxMessage/BoxMessage';


describe("Pruebas en <BoxMessage />", () => {

    const message = 'Mensaje prueba';

    test("Debe hacer match con el snapshot", () => {
        const { container } = render(<BoxMessage message='Hola' />)
        expect(container).toMatchSnapshot();
    });
    test("Debe mostrar el mensaje enviado por props en una etiqueta <p>", () => {
        render(<BoxMessage message={ message } />)
        expect(screen.getByRole('paragraph').innerHTML).toContain(message);
    })
})