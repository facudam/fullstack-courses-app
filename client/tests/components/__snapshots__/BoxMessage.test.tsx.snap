// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <BoxMessage /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="box-message"
    data-testid="test-message"
  >
    <p>
      Hola
    </p>
  </div>
</div>
`;
